exercises/allergies/.gitignore

```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
```

Comment:
What if I proposed that we can preserve all the benefits described, and avoid the potential problem of the student wondering "so am I supposed to remove this or not?" by editing the comment to be something like "Since this is a library, we leave Cargo.lock in the .gitignore, but if it were an executable we would remove it" ?

(I am OK with either solution here, but I am usually in favor of making suggestions even if they end up being crazy)